### ISSUE TITLE
Unable to train on very simple criteo like data

### ISSUE THREAD

**Issue by batrlatom** [shenweichen/DeepCTR#196]
Please refer to the [FAQ](https://deepctr-doc.readthedocs.io/en/latest/FAQ.html) in doc and search for the [related issues](https://github.com/shenweichen/DeepCTR/issues) before you ask the question.

**Describe the question(问题描述)**
I would like to ask why any model from DeepCTR is unable to train on my criteo like data?
I am using the same code from your example with simple change in the number of cols in sparse and dense features. The data has only one sparse and one dense column. Dense column is always the same for simplicity. Each odd number in I1 gives label 1, even number in I1 gives label 0. C1 could be anything

```
label, I1, C1
0, 0, a
1, 1, a
0, 2, a
1, 3, a
0, 4, a
...

```
output is:

Epoch 94/100
80/80 - 0s - loss: 0.6885 - binary_crossentropy: 0.6885 - val_loss: 0.7032 - val_binary_crossentropy: 0.7032
Epoch 95/100
80/80 - 0s - loss: 0.6885 - binary_crossentropy: 0.6885 - val_loss: 0.7032 - val_binary_crossentropy: 0.7032
Epoch 96/100
80/80 - 0s - loss: 0.6885 - binary_crossentropy: 0.6885 - val_loss: 0.7033 - val_binary_crossentropy: 0.7033
Epoch 97/100
80/80 - 0s - loss: 0.6885 - binary_crossentropy: 0.6885 - val_loss: 0.7033 - val_binary_crossentropy: 0.7033
Epoch 98/100
80/80 - 0s - loss: 0.6885 - binary_crossentropy: 0.6885 - val_loss: 0.7034 - val_binary_crossentropy: 0.7034
Epoch 99/100
80/80 - 0s - loss: 0.6885 - binary_crossentropy: 0.6885 - val_loss: 0.7035 - val_binary_crossentropy: 0.7035
Epoch 100/100
80/80 - 0s - loss: 0.6885 - binary_crossentropy: 0.6885 - val_loss: 0.7035 - val_binary_crossentropy: 0.7035
test LogLoss 0.7187
test AUC 0.3938

**Additional context**
I have chosen those data to model user behavior in time. For example the rule could be... user click on the ad on Monday, Wednesday, Friday, etc.

**Operating environment(运行环境):**
 - python version [e.g. 3.6]
Python 3.7
 - tensorflow version [e.g. 1.4.0,]
TF version 2.1.0
 - deepctr version [e.g. 0.7.1,]
latest

**Comment by TanTingyi** [comment]
The toydata you have provided is not suitable, each colum means a different field, it's meaningless when every element of **I1** is unique.

**Comment by TanTingyi** [comment]
You can try this code i have test.
`
    data_dict = dict(label=[0, 1]*5000, i1=list(range(10000)), c1=[0.5]*10000)
    data = pd.DataFrame(data_dict)


    sparse_features = ['i1']
    dense_features = ['c1']

    target = ['label']

    # 1.Label Encoding for sparse features,and do simple Transformation for dense features
    for feat in sparse_features:
        lbe = LabelEncoder()
        data[feat] = lbe.fit_transform(data[feat])
    mms = MinMaxScaler(feature_range=(0, 1))
    data[dense_features] = mms.fit_transform(data[dense_features])

    # 2.count #unique features for each sparse field,and record dense feature field name

    fixlen_feature_columns = [SparseFeat(feat, vocabulary_size=data[feat].nunique(),embedding_dim=4)
                           for i,feat in enumerate(sparse_features)] + [DenseFeat(feat, 1,)
                          for feat in dense_features]

    dnn_feature_columns = fixlen_feature_columns
    linear_feature_columns = fixlen_feature_columns

    feature_names = get_feature_names(linear_feature_columns + dnn_feature_columns)

    # 3.generate input data for model

    train, test = train_test_split(data, test_size=0.2)
    train_model_input = {name:train[name] for name in feature_names}
    test_model_input = {name:test[name] for name in feature_names}

    # 4.Define Model,train,predict and evaluate
    model = DeepFM(linear_feature_columns, dnn_feature_columns, task='binary')
    model.compile("adam", "binary_crossentropy",
                  metrics=['binary_crossentropy'], )

    history = model.fit(train_model_input, train[target].values,
                        batch_size=256, epochs=10, verbose=2, validation_split=0.2, )
    pred_ans = model.predict(test_model_input, batch_size=256)
    print("test LogLoss", round(log_loss(test[target].values, pred_ans), 4))
    print("test AUC", round(roc_auc_score(test[target].values, pred_ans), 4))
`

**Comment by batrlatom** [comment]
Thanks for the answer. Your code snippet was helpful with little upgrade ... repetition
```
    def modulo(n):
        # each odd week is click
        return n%4

    items = int(1e+3)
    data_dict = dict(label=[0, 1] * int(items/2), I1=list(map(modulo, list(range(items)))), C1=[0.5] * items)
```

**Comment by TanTingyi** [comment]
Hi, batrlatom, your code works well in my env : python 3.7.6 tf 2.1 , deepctr 0.7.4. 
You can upgrade your env and try again.
